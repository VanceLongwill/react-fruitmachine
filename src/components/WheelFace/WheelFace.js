// @flow
import React from 'react'
import styled from 'styled-components'

const Facet = styled.div`
  position: absolute;
  height: 100px;
  width: 100px;
  color: rgba(0, 0, 0, 0.9);
  border-radius: 5px;
  border: 10px solid rgb(191, 191, 191);
  background-color: ${p => p.color};
  transform: rotateX(${p => p.rotateX}deg) translateZ(${p => p.radius}px);
`

type Props = {
  color: string,
  radius: number,
  rotateX: number,
}

export function WheelFace(props: Props) {
  const { color, radius, rotateX } = props
  return <Facet color={color} radius={radius} rotateX={rotateX} />
}
