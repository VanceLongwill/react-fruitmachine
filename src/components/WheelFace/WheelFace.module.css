.face {
  position: absolute;
  left: 250px;
  width: 100px;
  height: 100px;
  opacity: 0.7;
  color: rgba(0,0,0,0.9);
  border-radius: 10px;
}

.inner {
  font-size: 36px;
  font-weight: bold;
  text-align: center;
  margin-top: 28px;
}

/*
.Animate {
  animation: spinWheel 2s ease infinite;
}

@keyframes spinWheel {
  100% {
    transform: translateY(-400%);
  }
}
*/
