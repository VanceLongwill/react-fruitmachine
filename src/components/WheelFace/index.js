import { WheelFace } from './WheelFace'
export default WheelFace
