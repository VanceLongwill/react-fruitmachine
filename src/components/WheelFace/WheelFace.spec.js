import React from 'react'
import { mount } from 'enzyme'

import { WheelFace } from './WheelFace'

describe('WheelFace', () => {
  it('should match snapshot', () => {
    expect(mount(<WheelFace color={'green'} radius={200} rotateX={20}/>)).toMatchSnapshot()
  })
})
