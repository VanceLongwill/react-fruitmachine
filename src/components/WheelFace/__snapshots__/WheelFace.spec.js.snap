// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WheelFace should match snapshot 1`] = `
<styled.div
  color="green"
  radius={200}
  rotateX={20}
>
  <div
    className="sc-bdVaJa bCPSZP"
    color="green"
    radius={200}
  />
</styled.div>
`;
